src/foam/nanos/medusa/ClusterConfigSupport.js

```diff
@@ -0,0 +1,1051 @@
+          return false;
+        }
+      }
+      getLogger().debug("hasNodeQuorum", "true");
+      return true;
+      `
+    },
+    {
+      documentation: 'Mediators to broadcast to. See ClusterConfigStatusDAO',
+      name: 'broadcastMediators',
+      class: 'FObjectArray',
+      of: 'foam.nanos.medusa.ClusterConfig',
+      javaFactory: `
+      ClusterConfig myConfig = getConfig(getX(), getConfigId());
+      long zone = myConfig.getZone() + 1;
+      if ( myConfig.getType() == MedusaType.NODE ) {
+        zone = myConfig.getZone();
+      }
+      // getLogger().debug("broadcastMediators", "zone", zone);
+      List<ClusterConfig> arr = (ArrayList) ((ArraySink) ((DAO) getX().get("localClusterConfigDAO"))
+        .where(
+          AND(
+            EQ(ClusterConfig.ZONE, zone),
+            OR(
+              EQ(ClusterConfig.TYPE, MedusaType.MEDIATOR),
+              EQ(ClusterConfig.TYPE, MedusaType.NERF)
+            ),
+            EQ(ClusterConfig.ENABLED, true),
+            EQ(ClusterConfig.REGION, myConfig.getRegion()),
+            EQ(ClusterConfig.REALM, myConfig.getRealm())
+          )
+        )
+        .select(new ArraySink())).getArray();
+      ClusterConfig[] configs = new ClusterConfig[arr.size()];
+      arr.toArray(configs);
+      return configs;
+      `
+    },
+    {
+      documentation: 'Any active region in realm.',
+      name: 'nextZone',
+      class: 'FObjectProperty',
+      of: 'foam.nanos.medusa.ClusterConfig',
+      javaFactory: `
+      ClusterConfig config = getConfig(getX(), getConfigId());
+      long zone = config.getZone();
+      while ( zone > 0 ) {
+        zone -= 1;
+        DAO dao = (DAO) getX().get("clusterConfigDAO");
+        dao = dao
+          .where(
+            AND(
+              EQ(ClusterConfig.ENABLED, true),
+              EQ(ClusterConfig.REALM, config.getRealm()),
+              EQ(ClusterConfig.REGION_STATUS, RegionStatus.ACTIVE),
+              EQ(ClusterConfig.STATUS, Status.ONLINE),
+              OR(
+                EQ(ClusterConfig.TYPE, MedusaType.MEDIATOR),
+                EQ(ClusterConfig.TYPE, MedusaType.NERF)
+              ),
+              EQ(ClusterConfig.ZONE, zone)
+            ));
+        if ( zone == 0 ) {
+          dao = dao.orderBy(foam.mlang.MLang.DESC(ClusterConfig.IS_PRIMARY));
+        } else {
+          dao = dao.orderBy(ClusterConfig.PING_TIME);
+        }
+        List<ClusterConfig> configs = ((ArraySink) dao.select(new ArraySink())).getArray();
+        if ( configs.size() > 0 ) {
+          // return configs.get(0);
+          // return configs.get(configs.size() -1);
+          ClusterConfig cfg = configs.get(0);
+          getLogger().info("nextZone", "configs", configs.size(), "selected", cfg.getId(), cfg.getZone(), cfg.getIsPrimary(), cfg.getPingTime());
+          for ( ClusterConfig c : configs ) {
+            getLogger().info("nextZone", "other", cfg.getId(), cfg.getZone(), cfg.getIsPrimary(), cfg.getPingTime());
```

Comment:
wondering should it log `c` instead of `cfg`?
```suggestion
          for ( ClusterConfig c : configs ) {
            getLogger().info("nextZone", "other", c.getId(), c.getZone(), c.getIsPrimary(), c.getPingTime());
```